### PR TITLE
openlineage, mysql: add OpenLineage support for MySQL

### PR DESCRIPTION
**Pull request by JDarDagran**
---
This PR adds OpenLineage support for MySqlOperator.

depends on: #31398

### CODE REVIEW

**Comment by pankajkoti** [comment]
@sunank200 could you please help review here?

File: airflow/providers/common/sql/hooks/sql.py [hunk 1]
```diff
@@ -516,3 +517,40 @@ def test_connection(self):
             message = str(e)
 
         return status, message
+
+    def get_database_info(self, connection):
```
Comment (sunank200) [inline airflow/providers/common/sql/hooks/sql.py]: missing doc-strings and annotations

File: airflow/providers/mysql/hooks/mysql.py [hunk 2]
```diff
@@ -298,3 +298,24 @@ def bulk_load_custom(
...
+            scheme=self.get_database_dialect(connection),
+            authority=self._get_authority(connection),
+            information_schema_columns=[
+                "table_schema",
+                "table_name",
+                "column_name",
+                "ordinal_position",
+                "column_type",
+            ],
+            normalize_name_method=lambda name: name.upper(),
```
Comment (sunank200) [inline airflow/providers/mysql/hooks/mysql.py:316]: is the name always supposed to be upper?

File: airflow/providers/openlineage/extractors/base.py [hunk 3]
```diff
@@ -35,6 +37,22 @@ class OperatorLineage:
     run_facets: dict[str, BaseFacet] = Factory(dict)
     job_facets: dict[str, BaseFacet] = Factory(dict)
 
+    @staticmethod
+    def merge(base: OperatorLineage, other: OperatorLineage) -> OperatorLineage:
+        """
```
Comment (sunank200) [inline airflow/providers/openlineage/extractors/base.py]: Should this be a separate static method? Shouldn't this be done while populating the lineage?

**Review by sunank200 — COMMENTED** [review]
Documentation for this change should be added accordingly

File: airflow/providers/openlineage/extractors/base.py (hunk 3 above)
Comment (JDarDagran) [reply airflow/providers/openlineage/extractors/base.py]: @sunank200 it comes from commit in #31398 that this PR is dependent on. I've deleted this method though there.

File: airflow/providers/mysql/hooks/mysql.py (hunk 2 above)
Comment (JDarDagran) [reply airflow/providers/mysql/hooks/mysql.py:316]: tbh case sensitivity across different SQL databases is a bit problematic but so far we assumed so for MySQL in OpenLineage and I would not want to break it in this PR

File: airflow/providers/common/sql/hooks/sql.py (hunk 1 above)
Comment (JDarDagran) [reply airflow/providers/common/sql/hooks/sql.py]: This method got changed in base PR https://github.com/apache/airflow/pull/31398.

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**